homeassistant/components/recorder/pool.py

```diff
@@ -1,34 +1,58 @@
 """A pool for sqlite connections."""
+import logging
 import threading
 
-from sqlalchemy.pool import NullPool, StaticPool
+from sqlalchemy.pool import NullPool, SingletonThreadPool
 
+from .const import DB_WORKER_PREFIX
 
-class RecorderPool(StaticPool, NullPool):
-    """A hybrid of NullPool and StaticPool.
+POOL_SIZE = 5
```

Comment:
This is the sqlalchemy default but we might not even need this many since it tends to be I/O bound.  I didn't lower though because there is always someone who has an ultra fast disk who could in theory benefit from it.  In practice its unlikely anything more than 1 or 2 is useful for most systems though.